quiz.go

```diff
@@ -21,31 +27,45 @@ func main() {
 
 	r := csv.NewReader(bufio.NewReader(file))
 
-	for {
-		record, err := r.Read()
-		if err == io.EOF {
-			break
-		}
-		if err != nil {
-			log.Fatal(err)
-			continue
-		}
+	go func() {
+		for {
+			record, err := r.Read()
+			if err == io.EOF {
+				finished <- true
+			}
+			if err != nil {
+				log.Fatal(err)
+			}
 
-		problem := record[0]
-		solution := record[1]
+			problem := record[0]
+			solution := record[1]
 
-		fmt.Printf("%v?\n", problem)
-		var response string
-		fmt.Scanln(&response)
+			fmt.Printf("%v?\n", problem)
+			var response string
+			fmt.Scanln(&response)
 
-		if response == solution {
-			fmt.Println("Correct!")
-			numCorrect++
-		} else {
-			fmt.Println("Incorrect :(")
-			numIncorrect++
+			if response == solution {
+				fmt.Println("Correct!")
+				numCorrect++
+			} else {
+				fmt.Println("Incorrect :(")
+				numIncorrect++
+			}
 		}
-	}
+	}()
+
+	timeLimit := time.After(30 * time.Second)
 
-	fmt.Printf(" Number correct: %v, number incorrect: %v\n", numCorrect, numIncorrect)
+	for {
```

Comment:
You don't need this for loop since you're only ever selecting either the time limit or finished once.

What tipped me off here was every case calling `os.Exit(0)`. [Check this out](https://play.golang.org/p/FZ3gTWRtnA)

Then you can move the `printResults` call to outside the `select` statement.